opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/DefaultSessionUtils.java

```diff
@@ -60,4 +84,60 @@
         }
     }
 
+    /**
+     * NOTE: This is mostly copied from org.springframework.transaction.support.TransactionTemplate#execute
+     * but adds the ability to specify our own transaction definition.
+     */
+    private <V> V executeWithTransactionDefinition(TransactionDefinition transactionDefinition, Supplier<V> supplier) {
+        final TransactionStatus status = transactionManager.getTransaction(transactionDefinition);
+        V result;
+        try {
+            result = supplier.get();
+        }
+        catch (RuntimeException ex) {
+            // Transactional code threw application exception -> rollback
+            rollbackOnException(status, ex);
+            throw ex;
+        }
+        catch (Error err) {
+            // Transactional code threw error -> rollback
+            rollbackOnException(status, err);
+            throw err;
+        }
+        catch (Throwable ex) {
```

Comment:
Wouldn't any `Supplier<V>` passed to this method cause a compile time error if it was throwing a non-runtime exception?

Maybe this is just a leftover from copying or I'm missing something. If its just a result of copying then its not a problem anyway, it just looks like it would be dead code to me.

Reply:
Suppliers can't throw checked exceptions AFAIK.

This code was copied verbatim.